token/program-2022/src/state.rs

```diff
@@ -290,15 +289,6 @@ fn unpack_coption_u64(src: &[u8; 12]) -> Result<COption<u64>, ProgramError> {
     }
 }
 
-// `spl_token_program_2022::extension::AccountType::Account` ordinal value
-const ACCOUNTTYPE_ACCOUNT: u8 = AccountType::Account as u8;
-impl GenericTokenAccount for Account {
-    fn valid_account_data(account_data: &[u8]) -> bool {
-        spl_token::state::Account::valid_account_data(account_data)
```

Comment:
Can we duplicate the `valid_account_data` logic (spl_token dependency) instead of removing this? That way we don't have to remember to re-add this either.

Reply:
Yeah sure, I'll duplicate the trait too, that way we'll just have to delete that trait declaration and impl the one from spl_token once we're ready.

And what's really neat, there won't be anything to re-add!